### PR TITLE
doc: Created a user guide for Thingy:91

### PR DESCRIPTION
**Pull request by umapraseeda**
[Working with Thingy_91 — nRF Connect SDK 1.1.99 documentation.pdf](https://github.com/NordicPlayground/fw-nrfconnect-nrf/files/4107652/Working.with.Thingy_91.nRF.Connect.SDK.1.1.99.documentation.pdf)
ref:https://projecttools.nordicsemi.no/jira/browse/NCSDK-3650
Created a new user guide for Thingy:91 including information
about building and programming sample application onto Thingy:91.
Added links to this user guide from the gsprogramming.rst (Getting
Started guide). Links to a new section `Programming a Thingy:91` in
programmer user guide in infocenter will be updated in this user guide.

Signed-off-by: Uma Praseeda <uma.praseeda@nordicsemi.no>

### CODE REVIEW

**Comment by umapraseeda** [comment]
@joakimtoe

**Comment by umapraseeda** [comment]
> I have some suggestions to improve the overall structure - the content looks really good, but I think reordering the sections will make it more user-friendly.
> Let's go through it tomorrow.

Thank you for the suggestions. As discussed, I have restructured the document and made some modifications. Attaching the latest version of the document.
[Working with Thingy_91 — nRF Connect SDK 1.1.99 documentation.pdf](https://github.com/NordicPlayground/fw-nrfconnect-nrf/files/4127946/Working.with.Thingy_91.nRF.Connect.SDK.1.1.99.documentation.pdf)

File: doc/nrf/ug_thingy91.rst [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+
+.. _thingy91_ug_intro:
+
+Working with Thingy:91
+######################
+
+The |NCS| provides support for developing on the Thingy:91 using the following inbuilt components:
+
+* nRF9160 System in Package (SiP) using the nRF9160 DK board (PCA10090)
+* nRF52840 System on Chip (SoC) using the nRF52840 DK board (PCA10056)
```
Comment (jtguggedal) [inline doc/nrf/ug_thingy91.rst]: I'm probably misunderstanding something here, just wondering why nRF9160 DK and nRF52840 DK are mentioned?

File: doc/nrf/ug_thingy91.rst [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+Building and programming from the source code
+*********************************************
+
+You can also program the Thingy:91 by using the images obtained by building the code in an |NCS| environment.
+
+To set up your system to be able to build a compatible firmware image, follow the `nRF Connect SDK Getting started guide`_.
+There are two build targets of interest for  Thingy:91 in |NCS|, namely:
+
+* ``nrf9160_pca20035ns``
```
Comment (jtguggedal) [inline doc/nrf/ug_thingy91.rst]: It's also possible to use the target `nrf9160_pca20035` for secure-only firmware. LTE/GPS can only be used with non-secure target.

File: doc/nrf/ug_thingy91.rst [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+      :start-after: build_SES_projimport_note_start
+      :end-before: build_SES_projimport_note_end
+
+#. To build the sample or an application:
+
+   a. Select your project in the Project Explorer.
+   #. From the menu, select **Build -> Build Solution**.
+      This builds the project.
+
+   The merged hex file, containing both the application and the SPM is located in ``\nrf\samples\nrf9160\at_client\build_nrf9160_pca20035\zephyr`` if you build the secure version and in ``\nrf\samples\nrf9160\at_client\build_nrf9160_pca20035ns\zephyr`` if you build the non-secure version.
```
Comment (jtguggedal) [inline doc/nrf/ug_thingy91.rst]: I'm not familiar with SES environment for building, but I think only the non-secure output should be valid for the AT client? BSD lib can't run from secure context

File: doc/nrf/gs_programming.rst [hunk 4]
```diff
@@ -11,11 +11,17 @@ the Nordic Edition of the SEGGER Embedded Studio (SES) IDE.
...
 .. _gs_programming_ses:
 
 Building with SES
 *****************
 
+.. build_SES_projimport_open_start
+
 Complete the following steps to build |NCS| projects with SES after :ref:`installing SEGGER Embedded Studio <installing_ses_win>` and :ref:`completing the first time setup <build_environment_win>`:
 
 1. Run the file :file:`bin/emStudio`.
```
Comment (OeySa) [inline doc/nrf/gs_programming.rst]: The sentence sounds like the user is supposed to run a script of some sort. Should it be "Start Segger Embedded Studio by running 'bin/emStudio' "?

File: doc/nrf/ug_thingy91.rst (hunk 1 above)
Comment (umapraseeda) [reply doc/nrf/ug_thingy91.rst]: Modified to
 * nRF9160 System in Package (SiP)
* nRF52840 System on Chip (SoC)

File: doc/nrf/ug_thingy91.rst (hunk 2 above)
Comment (umapraseeda) [reply doc/nrf/ug_thingy91.rst]: Placing this information in a table to include information about the secure and the non-secure versions. Also added the information ' LTE/GPS can only be used with non-secure target.' as a note.

File: doc/nrf/ug_thingy91.rst (hunk 3 above)
Comment (umapraseeda) [reply doc/nrf/ug_thingy91.rst]: Generalizing this as follows:
You can find the output of the build, which includes the merged HEX file containing both the application and the SPM, in the ``zephyr`` sub-folder in the build directory.

File: doc/nrf/gs_programming.rst (hunk 4 above)
Comment (umapraseeda) [reply doc/nrf/gs_programming.rst]: Modified as 
```suggestion
1. Start SEGGER Embedded Studio by running :file:`bin/emStudio`.
```

File: doc/nrf/ug_thingy91.rst [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+Thingy:91 has a multimode modem, which enables it to support automatic switching between LTE-M and NB-IoT.
+A built-in parameter in the Thingy:91 firmware determines whether the modem first attempts to connect in LTE-M or NB-IoT mode.
+If the modem fails to connect using this preferred mode within the default timeout period (10 minutes), the modem switches to the other mode.
+
+Modem firmware
+==============
+
+The functionality of the multimode modem of Thingy:91 is based on its firmware which is available as a pre-compiled binary.
+You can download the modem firmware from the `Thingy:91 product website (downloads)`_.
+The downloadable zip file contains the full firmware image and the firmware image files for updating the firmware.
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]: @joakimtoe Does the Thingy support incremental updates?

File: doc/nrf/ug_thingy91.rst [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+
+.. _precompiled_fw:
+
+Programming pre-compiled firmware images
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]: "precompiled" according to our terms list (occurs in more places)

File: doc/nrf/ug_thingy91.rst [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+* Restoring the firmware to its initial image
+
+* Updating the application firmware to a newly released version
+
+* Updating the modem firmware
+
+Downloading pre-compiled firmware images
+========================================
+
+To obtain pre-compiled firmware images for firmware update, perform the following steps:
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]:
```suggestion
To obtain precompiled firmware images for updating the firmware, perform the following steps:
```
or "for firmware updates"

File: doc/nrf/ug_thingy91.rst [hunk 8]
```diff
@@ -0,0 +1,262 @@
...
+
+	1. Go to the `Thingy:91 product description`_ page and under the **Downloads** tab, navigate to **Precompiled firmware**.
+	#. Download and extract the latest Thingy:91 firmware package.
+	#. Check the ``CONTENTS.txt`` file in the extracted folder for the location and names of the different firmware images.
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]:
```suggestion
	#. Check the :file:`CONTENTS.txt` file in the extracted folder for the location and names of the different firmware images.
```

File: doc/nrf/ug_thingy91.rst [hunk 9]
```diff
@@ -0,0 +1,262 @@
...
+
+.. _programming_usb:
+
+Quick programming of pre-compiled firmware images
+=================================================
+
+When you have the pre-compiled firmware images ready, you can directly program the images onto the Thingy:91 using the `nRF Connect Programmer`_ app, which is implemented as part of the `nRF Connect for Desktop`_.
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]:
```suggestion
When you have the pre-compiled firmware images ready, you can directly program the images onto the Thingy:91 using the `nRF Connect Programmer`_ app, which is available in  `nRF Connect for Desktop`_.
```

File: doc/nrf/ug_thingy91.rst [hunk 10]
```diff
@@ -0,0 +1,262 @@
...
+
+This is a direct USB method, making use of the MCUboot feature and the inbuilt serial recovery mode of Thingy:91.
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]: Not sure what we mean with "direct USB method".
Maybe:
When programming with nRF Connect Programmer, you connect the Thingy:91 directly to your PC through USB.
This method makes use of :doc:`mcuboot:index` and the inbuilt ...

File: doc/nrf/ug_thingy91.rst [hunk 11]
```diff
@@ -0,0 +1,262 @@
...
+You can program the Thingy:91 either on the nRF9160 SiP or on the nRF52840 SoC component.
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]:
```suggestion
You can program either the nRF9160 SiP or the nRF52840 SoC component on the Thingy:91.
```

File: doc/nrf/ug_thingy91.rst [hunk 12]
```diff
@@ -0,0 +1,262 @@
...
+
+For the detailed procedure for programming a Thingy:91 using nRF Connect Programmer, see the following: (links to the thingy91 programming sections in nRF connect programmer userguide in infocenter to be added here).
+
+Updating modem firmware
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]:
```suggestion
Updating the modem firmware
```

File: doc/nrf/ug_thingy91.rst [hunk 13]
```diff
@@ -0,0 +1,262 @@
...
+See (link to the thingy91 modem update section in nRF connect programmer userguide in infocenter to be added here) for the detailed steps to update the modem firmware.
+
+.. _building_pgming:
+
+Building and programming from the source code
+*********************************************
+
+You can also program the Thingy:91 by using the images obtained by building the code in an |NCS| environment.
+
+To set up your system to be able to build a compatible firmware image, follow the `nRF Connect SDK Getting started guide`_.
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]: ... the :ref:`getting_started`.

File: doc/nrf/ug_thingy91.rst [hunk 14]
```diff
@@ -0,0 +1,262 @@
...
+The build targets of interest for  Thingy:91 in |NCS|, are as follows:
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]:
```suggestion
The build targets of interest for Thingy:91 in |NCS| are as follows:
```

File: doc/nrf/ug_thingy91.rst [hunk 15]
```diff
@@ -0,0 +1,262 @@
...
+
++---------------+--------------------------------------------------+
+|Component      |  Build Target                                    |
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]:
```suggestion
|Component      |  Build target                                    |
```

File: doc/nrf/ug_thingy91.rst [hunk 16]
```diff
@@ -0,0 +1,262 @@
...
++===============+==================================================+
+|nRF9160 SiP    |``nrf9160_pca20035`` for the secure version       |
+|               |                                                  |
+|               |``nrf9160_pca20035ns`` for the non-secure version |
++---------------+--------------------------------------------------+
+|nRF52840 SoC   |``nrf52840_pca20035``                             |
++---------------+--------------------------------------------------+
+
+You should use the build target ``nrf9160_pca20035`` or ``nrf9160_pca20035ns``, when building the application code for the nRF9160 SiP and the build target ``nrf52840_pca20035``, when building the application code for the onboard nRF52840 SoC.
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]:
```suggestion
You should use the build target ``nrf9160_pca20035`` or ``nrf9160_pca20035ns`` when building the application code for the nRF9160 SiP and the build target ``nrf52840_pca20035`` when building the application code for the onboard nRF52840 SoC.
```

File: doc/nrf/ug_thingy91.rst [hunk 17]
```diff
@@ -0,0 +1,262 @@
...
+|app_update.bin| MCUboot compatible image, binary format|* Using ``mcumgr`` command line tool   |
+|              |                                        |* For FOTA updates                     |
++--------------+----------------------------------------+---------------------------------------+
+
+There are multiple methods of programming a sample or application onto a Thingy:91.
+You can choose the method based on the availability or absence of an external debug probe to program.
+
+.. note::
+
+   In the event of not having an external debug probe to program the Thingy:91, you can directly program by :ref:`using the USB (MCUboot) method and nRF Connect Programmer<programming_usb>`.
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]:
```suggestion
   If you do not have an external debug probe available to program the Thingy:91, you can directly program by :ref:`using the USB (MCUboot) method and nRF Connect Programmer<programming_usb>`.
```

File: doc/nrf/ug_thingy91.rst [hunk 18]
```diff
@@ -0,0 +1,262 @@
...
+   The ``app_signed.hex`` firmware image file is used for the update in this scenario.
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]:
```suggestion
   In this scenario, use the :file:`app_signed.hex` firmware image file.
```

File: doc/nrf/ug_thingy91.rst [hunk 19]
```diff
@@ -0,0 +1,262 @@
...
+
+.. include:: gs_programming.rst
+   :start-after: build_SES_projimport_open_start
+   :end-before: build_SES_projimport_open_end
+
+
+.. figure:: images/ses_thingy_configuration.png
+   :alt: Opening the AT Client sample
+
+   Opening the AT Client sample for the nrf9160_pca20035ns build target
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]: Why are we using the AT Client sample here instead of the asset tracker?

File: doc/nrf/ug_thingy91.rst [hunk 20]
```diff
@@ -0,0 +1,262 @@
...
+      :start-after: build_SES_projimport_note_start
+      :end-before: build_SES_projimport_note_end
+
+#. To build the sample or an application:
+
+   a. Select your project in the Project Explorer.
+   #. From the menu, select **Build -> Build Solution**.
+      This builds the project.
+
+   You can find the output of the build, which includes the merged HEX file containing both the application and the SPM, in the ``zephyr`` sub-folder in the build directory.
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]:
```suggestion
   You can find the output of the build, which includes the merged HEX file containing both the application and the SPM, in the ``zephyr`` subfolder in the build directory.
```

File: doc/nrf/ug_thingy91.rst [hunk 21]
```diff
@@ -0,0 +1,262 @@
...
+   The *Board Directory* folder can be found in the following location: ``ncs/nrf/boards/arm``.
+
+
+4. Click **OK** to import the project into SES. You can now work with the project in the IDE.
+
+   .. include:: gs_programming.rst
+      :start-after: build_SES_projimport_note_start
+      :end-before: build_SES_projimport_note_end
+
+#. To build the sample or an application:
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]:
```suggestion
#. To build the sample or application:
```

File: doc/nrf/ug_thingy91.rst [hunk 22]
```diff
@@ -0,0 +1,262 @@
...
+   #. From the menu, select **Build -> Build Solution**.
+      This builds the project.
+
+   You can find the output of the build, which includes the merged HEX file containing both the application and the SPM, in the ``zephyr`` sub-folder in the build directory.
+
+.. prog_extdebugprobe_start
+
+6. To program the sample or application:
+
+   a. Set the Thingy:91 SWD selection switch (**SW2**) to **nRF91** or **nRF52** depending on whether you have to program the nRF9160 SiP or the nRF52840 SoC component.
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]:
```suggestion
   a. Set the Thingy:91 SWD selection switch (**SW2**) to **nRF91** or **nRF52** depending on whether you want to program the nRF9160 SiP or the nRF52840 SoC component.
```

File: doc/nrf/ug_thingy91.rst [hunk 23]
```diff
@@ -0,0 +1,262 @@
...
+.. prog_extdebugprobe_end
+..
+
+   e. In SES, select **Target -> Connect J-Link**.
+   #. Select **Target -> Download zephyr/merged.hex** to program the sample or application onto Thingy:91.
+   #. The device will reset and run the programmed sample or application.
+
+.. _build_pgm_cmdline:
+
+Building and programming using Command line
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]:
```suggestion
Building and programming on the command line
```

File: doc/nrf/ug_thingy91.rst [hunk 24]
```diff
@@ -0,0 +1,262 @@
...
+===========================================
+
+Complete the command line build setup (see the instructions for :ref:`Windows<build_environment_cli_win>` and :ref:`Linux<build_environment_cli_linux>`) before you start building |NCS| projects on the command line.
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]: Do we have instructions for macOS as well?

File: doc/nrf/ug_thingy91.rst [hunk 25]
```diff
@@ -0,0 +1,262 @@
...
+
+To build and program the source code from the command line, complete the following steps:
+
+1. Open a terminal window.
+#. Go to the specific sample or application directory.
+   For example, the folder path is ``ncs/nrf/applications/asset_tracker`` when building the source code for the :ref:`asset_tracker` application on the nRF9160 SiP component and ``ncs/nrf/samples/usb/usb_uart_bridge`` when building the source code for the :ref:`usb_uart_bridge_sample` sample on the nRF52840 SoC component.
+
+#. Make sure that you have the latest version of |NCS| repository by pulling the |NCS| repository, `fw-nrfconnect-nrf`_ on GitHub using the command:
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]: @joakimtoe This sentence only makes sense if we want people to work on master. Do we? Then we should also point this out.

File: doc/nrf/ug_thingy91.rst [hunk 26]
```diff
@@ -0,0 +1,262 @@
...
+#. To build the sample or application code, run the ``west build`` command as follows:
+
+   .. parsed-literal::
+      :class: highlight
+
+	  west build -b *board_name* -d *destination_directory_name*
+
+   .. note::
+
+      The parameter *board_name* should be ``nrf9160_pca20035`` or ``nrf9160_pca20035ns`` if building on the nRF9160 SiP component and ``nrf52840_pca20035`` if building on the nRF52840 SoC component.
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]:
```suggestion
      The parameter *board_name* should be ``nrf9160_pca20035`` or ``nrf9160_pca20035ns`` if building for the nRF9160 SiP component and ``nrf52840_pca20035`` if building for the nRF52840 SoC component.
```

File: doc/nrf/ug_thingy91.rst [hunk 27]
```diff
@@ -0,0 +1,262 @@
...
+      west update
+
+#. To build the sample or application code, run the ``west build`` command as follows:
+
+   .. parsed-literal::
+      :class: highlight
+
+	  west build -b *board_name* -d *destination_directory_name*
+
+   .. note::
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]: Add this as normal text and not as note (to avoid two consecutive notes).

File: doc/nrf/ug_thingy91.rst [hunk 28]
```diff
@@ -0,0 +1,262 @@
...
+
+	  The parameter *destination_directory_name* can be used to optionally specify the destination directory in the west command.
+	  Unless a *destination_directory_name* is specified, the build files are automatically generated in ``build/zephyr/``.
+
+.. include:: ug_thingy91.rst
+   :start-after: prog_extdebugprobe_start
+   :end-before: prog_extdebugprobe_end
+..
+
+   e. Navigate to the directory of the sample or application for which you have built the images.
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]: You should already be in there, right?

File: doc/nrf/ug_thingy91.rst [hunk 29]
```diff
@@ -0,0 +1,262 @@
+.. _ug_thingy91:
+
+.. _thingy91_ug_intro:
+
+Working with Thingy:91
+######################
+
+The |NCS| provides support for developing on the Thingy:91 using the following inbuilt components:
```
[the remark this answers is not in the document]
Comment (umapraseeda) [reply doc/nrf/ug_thingy91.rst]: Removed the list and retained the following sentence at the end of the section: The |NCS| provides support for developing applications on the Thingy:91.

File: doc/nrf/ug_thingy91.rst (hunk 10 above)
Comment (umapraseeda) [reply doc/nrf/ug_thingy91.rst]: changed to : In this method, the Thingy:91 is connected directly to your PC through USB. This method makes use of ....

File: doc/nrf/ug_thingy91.rst (hunk 19 above)
Comment (umapraseeda) [reply doc/nrf/ug_thingy91.rst]: changed the screenshot and the example to 'Opening the Asset tracker application'

File: doc/nrf/ug_thingy91.rst (hunk 28 above)
Comment (umapraseeda) [reply doc/nrf/ug_thingy91.rst]: Deleted this step

File: doc/nrf/ug_thingy91.rst (hunk 5 above)
Comment (joakimtoe) [reply doc/nrf/ug_thingy91.rst]: I would say something like: "The downloadable zip file contains both the full firmware images used with an external debug probe and the firmware images used for updating the firmware through the built in bootloader"

File: doc/nrf/ug_thingy91.rst (hunk 25 above)
Comment (joakimtoe) [reply doc/nrf/ug_thingy91.rst]: Ahh, you are correct. Its up to the developer to decide if he wants to base his work on a tag or master. I suggest we guide in both scenarios.

File: doc/nrf/ug_thingy91.rst [hunk 30]
```diff
@@ -0,0 +1,262 @@
...
+.. note::
+
+   LTE/GPS features can only be used with non-secure target.
+
+The table below shows the different types of build files that are generated and the different scenarios in which they are used:
+
++--------------+----------------------------------------+---------------------------------------+
+| File         | File format                            | Programming scenario                  |
++==============+========================================+=======================================+
+|merged.hex    | Full image, HEX format                 | Using debug probe                     |
```
Comment (joakimtoe) [inline doc/nrf/ug_thingy91.rst]: Using external debug probe and nRF Connect Programmer

File: doc/nrf/ug_thingy91.rst [hunk 31]
```diff
@@ -0,0 +1,262 @@
...
++--------------+----------------------------------------+---------------------------------------+
+|app_signed.hex| MCUboot compatible image, HEX format   | Using nRF Connect Programmer          |
```
Comment (joakimtoe) [inline doc/nrf/ug_thingy91.rst]: Using the built in bootloader and nRF Connect Programmer

File: doc/nrf/ug_thingy91.rst [hunk 32]
```diff
@@ -0,0 +1,262 @@
...
++--------------+----------------------------------------+---------------------------------------+
+|app_update.bin| MCUboot compatible image, binary format|* Using ``mcumgr`` command line tool   |
```
Comment (joakimtoe) [inline doc/nrf/ug_thingy91.rst]: Using the built in bootloader and ``mcumgr`` command line tool

File: doc/nrf/ug_thingy91.rst (hunk 24 above)
Comment (joakimtoe) [reply doc/nrf/ug_thingy91.rst]: Yes, https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/gs_ins_mac.html

File: doc/nrf/ug_thingy91.rst (hunk 30 above)
Comment (umapraseeda) [reply doc/nrf/ug_thingy91.rst]: Modified.

File: doc/nrf/ug_thingy91.rst (hunk 31 above)
Comment (umapraseeda) [reply doc/nrf/ug_thingy91.rst]: Modified.

File: doc/nrf/ug_thingy91.rst (hunk 32 above)
Comment (umapraseeda) [reply doc/nrf/ug_thingy91.rst]: Modified.

File: doc/nrf/ug_thingy91.rst (hunk 24 above)
Comment (umapraseeda) [reply doc/nrf/ug_thingy91.rst]: Added this.

File: doc/nrf/ug_thingy91.rst (hunk 5 above)
Comment (umapraseeda) [reply doc/nrf/ug_thingy91.rst]: Modified.

File: doc/nrf/ug_thingy91.rst (hunk 25 above)
Comment (umapraseeda) [reply doc/nrf/ug_thingy91.rst]: Replaced the existing text with : 
Make sure that you have the required version of |NCS| repository by pulling the |NCS| repository, `fw-nrfconnect-nrf`_ on GitHub using the procedures described in :ref:`dm-wf-get-ncs` and :ref:`dm-wf-update-ncs`.

File: doc/nrf/ug_thingy91.rst [hunk 33]
```diff
@@ -0,0 +1,262 @@
...
+Thingy:91 has a multimode modem, which enables it to support automatic switching between LTE-M and NB-IoT.
+A built-in parameter in the Thingy:91 firmware determines whether the modem first attempts to connect in LTE-M or NB-IoT mode.
+If the modem fails to connect using this preferred mode within the default time-out period (10 minutes), the modem switches to the other mode.
+
+Modem firmware
+==============
+
+The functionality of the multimode modem of Thingy:91 is based on its firmware, which is available as a precompiled binary.
+You can download the modem firmware from the `Thingy:91 product website (downloads)`_.
+The downloadable zip file contains both the full firmware images used with an external debug probe and the firmware images used for updating the firmware through the built-in bootloader.
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]:
```suggestion
The downloadable zip file contains both the full firmware images that can be programmed using an external debug probe and the firmware images that can be used for updating the firmware through the built-in bootloader.
```

File: doc/nrf/ug_thingy91.rst [hunk 34]
```diff
@@ -0,0 +1,262 @@
...
+Quick programming of precompiled firmware images
+=================================================
+
+When you have the precompiled firmware images ready, you can directly program the images onto the Thingy:91 using the `nRF Connect Programmer`_ app, which is available in `nRF Connect for Desktop`_.
+
+In this method, the Thingy:91 is connected directly to your PC through USB.
+This method makes use of the :doc:`mcuboot:index` feature and the inbuilt serial recovery mode of Thingy:91.
+You can program either the nRF9160 SiP or the nRF52840 SoC component on the Thingy:91.
+
+For the detailed procedures for programming a Thingy:91 using nRF Connect Programmer, see the following:
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]:
```suggestion
For the detailed procedures for programming a Thingy:91 using nRF Connect Programmer, see the following documentation:
```

File: doc/nrf/ug_thingy91.rst [hunk 35]
```diff
@@ -0,0 +1,262 @@
...
+   :alt: Opening the Asset tracker application
+
+   Opening the Asset tracker application for the nrf9160_pca20035ns build target
+
+.. note::
+
+   The *Board Directory* folder can be found in the following location: ``ncs/nrf/boards/arm``.
+
+
+4. Click **OK** to import the project into SES. You can now work with the project in the IDE.
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]: Line break.

File: doc/nrf/ug_thingy91.rst [hunk 36]
```diff
@@ -0,0 +1,262 @@
...
+
+Complete the command line build setup (see the instructions for :ref:`Windows<build_environment_cli_win>`, :ref:`Linux<build_environment_cli_linux>`, and :ref:`macOS<build_environment_cli_mac>`) before you start building |NCS| projects on the command line.
+
+To build and program the source code from the command line, complete the following steps:
+
+1. Open a terminal window.
+#. Go to the specific sample or application directory.
+   For example, the folder path is ``ncs/nrf/applications/asset_tracker`` when building the source code for the :ref:`asset_tracker` application on the nRF9160 SiP component and ``ncs/nrf/samples/usb/usb_uart_bridge`` when building the source code for the :ref:`usb_uart_bridge_sample` sample on the nRF52840 SoC component.
+
+#. Make sure that you have the required version of |NCS| repository by pulling the |NCS| repository, `fw-nrfconnect-nrf`_ on GitHub using the procedures described in :ref:`dm-wf-get-ncs` and :ref:`dm-wf-update-ncs`.
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]:
```suggestion
#. Make sure that you have the required version of the |NCS| repository by pulling the |NCS| repository, `fw-nrfconnect-nrf`_ on GitHub using the procedures described in :ref:`dm-wf-get-ncs` and :ref:`dm-wf-update-ncs`.
```

File: doc/nrf/ug_thingy91.rst [hunk 37]
```diff
@@ -0,0 +1,262 @@
...
+
+#. To get the rest of the dependencies, run the ``west`` command as follows:
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]:
```suggestion
#. To get the rest of the dependencies, run the ``west update`` command as follows:
```

Either this or just "``west`` command" in the following step - both works, it should just be consistent. :)

File: doc/nrf/ug_thingy91.rst [hunk 38]
```diff
@@ -0,0 +1,262 @@
+.. _ug_thingy91:
+
+.. _thingy91_ug_intro:
+
+Working with Thingy:91
+######################
+
+Thingy:91 is a battery-operated prototyping platform for cellular IoT systems, designed especially for asset tracking applications and environmental monitoring.
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst]:
```suggestion
Nordic Thingy:91 is a battery-operated prototyping platform for cellular IoT systems, designed especially for asset tracking applications and environmental monitoring.
```
Let's use the full name on the first occurrence.

File: doc/nrf/gs_programming.rst [hunk 39]
```diff
@@ -67,7 +81,7 @@ Complete the following steps to build |NCS| projects with SES after :ref:`instal
       Put the **SW5** switch (marked debug/prog) in the **NRF91** position to program the main controller, or in the **NRF52** position to program the board controller.
       See the `Device programming section in the nRF9160 DK User Guide`_ for more information.
 
-   To build and program an application:
+#. To build and program an application:
```
[the remark this answers is not in the document]
Comment (umapraseeda) [reply doc/nrf/gs_programming.rst]: Removed.

File: doc/nrf/ug_thingy91.rst (hunk 33 above)
Comment (umapraseeda) [reply doc/nrf/ug_thingy91.rst]: Modified.

File: doc/nrf/ug_thingy91.rst (hunk 34 above)
Comment (umapraseeda) [reply doc/nrf/ug_thingy91.rst]: Added.

File: doc/nrf/ug_thingy91.rst (hunk 35 above)
Comment (umapraseeda) [reply doc/nrf/ug_thingy91.rst]: Added.

File: doc/nrf/ug_thingy91.rst (hunk 36 above)
Comment (umapraseeda) [reply doc/nrf/ug_thingy91.rst]: Added.

File: doc/nrf/ug_thingy91.rst (hunk 38 above)
Comment (umapraseeda) [reply doc/nrf/ug_thingy91.rst]: Added.

File: doc/nrf/ug_thingy91.rst (hunk 37 above)
Comment (umapraseeda) [reply doc/nrf/ug_thingy91.rst]: Modified

**Pull request merged**